proxy/proxy.go

```diff
@@ -437,7 +437,12 @@ func sendDirectResponse(w http.ResponseWriter, r *http.Request,
 		w.Header().Set(hdrGrpcStatus, strconv.Itoa(int(codes.Internal)))
 		w.Header().Set(hdrGrpcMessage, errInfo)
 
-		w.WriteHeader(statusCode)
+		// As per the gRPC spec, we need to send a 200 OK status code
+		// even if the request failed. The Grpc-Status and Grpc-Message
+		// header fields are enough to inform any gRPC compliant client
+		// about the error. See:
+		// https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#responses
+		w.WriteHeader(http.StatusOK)
```

Comment:
Will this be backwards compatible with the prior users of this package? I wonder if we should consider making this into a sub-module?

Reply:
Yes, this should be backward compatible, since a gRPC client only looks at the `Grpc-Status` and `Grpc-Message` anyway. AFAIK this was always true, but newer clients just started validating the HTTP status code as well.
But this is what I aim to confirm with an itest and an older client.